v1.0/deploy-cockroachdb-to-production.md

```diff
@@ -0,0 +1,424 @@
+
+However, CockroachDB supports using any valid certificates, such as those generated by OpenSSL.
+
+Regardless of which option you choose, you'll need the following files:
+
+- A certificate authority (CA) certificate and key, used to sign all of the other certificates
+- A separate certificate and key for each node in your deployment, with the common name `node`
+- A separate certificate and key for each client and user you want to connect to your nodes, with the common name set to the username. The default user is `root`.
+  
+  Alternatively, CockroachDB does [support password authentication](create-and-manage-users.html#secure-cluster), though we typically recommend using client certificates instead.
+
+### Deploying Certificates
+
+#### Nodes
+
+Each node requires local copies of the following:
+
+- Your certificate authority (CA) certificate to validate the signature of all other certificates
+- A node key and certificate signed by your CA with the common name of `node`
+- To run commands locally on nodes, you'll also need to include local copies certificate and key files for a user with the necessary permissions. By default, this is the `root` user.
+
+#### Clients
+
+Each client (e.g., your application or load balancer) requires local copies of the following:
+
+- Your certificate authority (CA) certificate to validate the signature of all other certificates
+- A node key and certificate signed by your CA with the common name set to the username. The default user is `root`.
+
+## Hardware Recommendations
+
+Minimum recommendations:
+
+- For a replicated cluster, use at least 3 nodes to ensure availability if a single node fails (see [Cluster Topology](#cluster-topology) for more details).
+- Each node should have sufficient CPU, RAM, network, and storage capacity to handle your workload, but the bare minimum is 1 CPU and 2 GB of RAM per node. More data, complex workloads, higher concurrency, and faster performance require additional resources.
+
+For best performance:
+
+- Use SSDs over HDDs.
+- Use larger/more powerful nodes. Adding more CPU is usually more beneficial than adding more RAM.
+
+For best resiliency:
+
+- Use many smaller nodes instead of fewer larger ones. Recovery from a failed node is faster when data is spread across more nodes.
+- Use [zone configs](configure-replication-zones.html) to increase the replication factor from 3 (the default) to 5. You can do this for the [entire cluster](configure-replication-zones.html#edit-the-default-replication-zone) or for specific [databases](configure-replication-zones.html#create-a-replication-zone-for-a-database) or [tables](configure-replication-zones.html#create-a-replication-zone-for-a-table).
+
+## Clock Synchronization
+
+{% include faq/clock-synchronization.html %}
+
+## Cache Size
+
+If you run multiple applications on the same machine as a CockroachDB node, you might consider manually setting the cache size instead of using the default 25% of available memory.
```

Comment:
it can also apply if running other memory-intensive processes. let's say a java app on the same machine.